apps/src/dance/lab2/ProgramExecutor.ts

```diff
@@ -102,7 +104,7 @@ export default class ProgramExecutor {
   /**
    * Preview the program. Compiles student code and calls on the native API to draw a frame.
    */
-  async preview() {
+  async preview(songMetadata: SongMetadata) {
```

Comment:
If we want to move legacy dance onto lab2, we're going to need a static preview here, yeah? I'm fine removing the static preview code here for now (we can readd when we cross that bridge), but could we call this `livePreview` (and I'll rename the legacy one to `staticPreview`) to distinguish between them for future users?

Reply:
Yep, good call, will update!